moto/awslambda/models.py

```diff
@@ -489,9 +489,16 @@ def __init__(
                 ).hexdigest()
                 self.code_size = 0
             else:
-                uri, tag = self.code["ImageUri"].split(":")
+                if "@" in self.code["ImageUri"]:
```

Comment:
Would you mind adding a test for this usecase? It seems pretty straightforward, but still good to verify that creating a function succeeds using this config.